index.html

```diff
@@ -1390,33 +1455,77 @@ <h2>Dereferencing the Resource</h2>
 					</ol>
 				</li>
 
-				<li>Otherwise, if the <var>input <a>DID URL</a></var> contains a <a>DID path</a> and/or <a>DID query</a>:
-				<pre class="example nohighlight">did:example:1234/custom/path?customquery</pre>
+				<li>If the <var>input <a>DID URL</a></var> contains a <a>DID path</a>:
+				<pre class="example nohighlight">did:example:1234/path/to/resource</pre>
 				<ol class="algorithm">
-					<li>The applicable <a>DID method</a> MAY specify how to dereference
-						the <a>DID path</a> and/or <a>DID query</a> of the <var>input <a>DID URL</a></var>.
-						<pre class="example nohighlight">did:example:1234/resources/1234</pre>
+					<li>Extract the path component from the <a>DID URL</a>.</li>
+					<li>
+						If the <a>DID Method</a> specifies its own path handling
+						algorithm, it MUST be applied, and the rest of this
+						algorithm skipped. Creators of <a>DID method</a>
+						specifications are encouraged to use this algorithm where
+						possible to increase interoperability.
 					</li>
-					<li>An extension specification MAY specify how to dereference
-						the <a>DID path</a> of the <var>input <a>DID URL</a></var>.
-						<pre class="example nohighlight">did:example:1234/whois</pre>
+					<li>
+						Collect from the resolved <a>DID Document</a> a list of <code>service</code>
+						objects with <code>type</code> values that the DID
+						Resolver supports for path handling. At minimum, that
+						will include the <code>type</code> values defined in
+						this specification (<a
+						href="#did-service-types">here</a>), plus any
+						<code>type</code> values that are DID Method-specific or
+						DID Resolution Extensions [[?DID-EXTENSIONS-RESOLUTION]]
+						supported by the DID Resolver. Such
+						<code>service</code> objects MUST have a
+						<code>prefix</code> attribute in addition to the
+						<code>type</code> attribute.
 					</li>
-					<li>An extension specification MAY specify how to dereference
-						the <a>DID query</a> of the <var>input <a>DID URL</a></var>.
-						<pre class="example nohighlight">did:example:1234?transformKey=JsonWebKey</pre>
+					<li>
+						Find the service whose <code>prefix</code> value is a
+						case-sensitive string prefix of the extracted path.
+						Among all matching services, select the one with the
+						longest matching prefix. That service MUST be used for
+						path resolution.
+						<ol class="algorithm">
+							<li>If two or more <code>services</code> have the
+							same longest matching prefix, the resolver MUST fail
+							resolution with an appropriate error.</li>
+							<li>If no matching service is found, the resolver
+							MUST return an error indicating that no applicable
```

Comment:
Again, lets define an error for this in https://w3c.github.io/did-resolution/#errors